src/noshow.jl

```diff
@@ -0,0 +1,60 @@
+
+"""
+    NoShow(layer)
+    NoShow(string, layer)
+
+This alters printing (for instance at the REPL prompt) to let you hide the complexity
+of some part of a Flux model. It has no effect on the actual running of the model.
+
+By default it prints `NoShow(...)` instead of the given layer.
+If you provide a string, it prints that instead -- it can be anything,
+but it may make sense to print the name of a function which will
```

Comment:
I thought a bit about allowing the user to specify the function (and its args) and incorporating something like https://github.com/JuliaLang/julia/blob/197180d8589ad14fc4bc4c23782b76739c4ec5a4/base/show.jl#L522 to make this more robust when a callable function is what is desired. I don't think it is worth the implementation complexity, and could also easily be added later if we really wanted it, so just noting for posterity.